src/psyclone/tests/psyir/symbols/symboltable_test.py

```diff
@@ -1643,3 +1646,232 @@ def test_rename_symbol():
         schedule_symbol_table.rename_symbol(symbol, "aRRay")
     assert ("The name argument of rename_symbol() must not already exist in "
             "this symbol_table instance, but 'aRRay' does." in str(err.value))
+
+
+def test_resolve_imports(fortran_reader, tmpdir):
+    ''' Tests the SymbolTable resolve_imports method. '''
+    with open(os.path.join(str(tmpdir), "a_mod.f90"), "w") as module:
+        module.write('''
+        module a_mod
+            integer :: a_1, a_2
+            integer :: b_1  ! Name clash but it is not imported
+        end module a_mod
+        ''')
+    with open(os.path.join(str(tmpdir), "b_mod.f90"), "w") as module:
+        module.write('''
+        module b_mod
+            integer, parameter :: b_1 = 10
+            integer, save, pointer :: b_2
+            integer :: not_used1
```

Comment:
Please could you add a private variable to this module and extend the test accordingly.

Reply:
Instead, I added a new test to better check the accessibility statements are respected, does it cover what you had in mind?